src/sinks/nats/config.rs

```diff
@@ -130,8 +136,56 @@ impl NatsSinkConfig {
 
         options.connect(&self.url).await.context(ConnectSnafu)
     }
+
+    pub(super) async fn publisher(&self) -> Result<NatsPublisher, NatsError> {
+        let connection = self.connect().await?;
+
+        match self.jetstream {
+            true => Ok(NatsPublisher::JetStream(async_nats::jetstream::new(
+                connection,
+            ))),
+            false => Ok(NatsPublisher::Core(connection)),
+        }
+    }
 }
 
 async fn healthcheck(config: NatsSinkConfig) -> crate::Result<()> {
     config.connect().map_ok(|_| ()).map_err(|e| e.into()).await
 }
+
+pub enum NatsPublisher {
+    Core(async_nats::Client),
+    JetStream(async_nats::jetstream::Context),
+}
+
+impl NatsPublisher {
+    pub(super) async fn publish<S: async_nats::subject::ToSubject>(
+        &self,
+        subject: S,
+        payload: Bytes,
+    ) -> Result<(), NatsError> {
+        match self {
+            NatsPublisher::Core(client) => {
+                client
+                    .publish(subject, payload)
+                    .await
+                    .map_err(|e| NatsError::PublishError {
+                        source: Box::new(e),
+                    })?;
+                client.flush().await.map_err(|e| NatsError::PublishError {
```

Comment:
In current version of async-nats - we no longer need to manually flush messages.
You can drop this call.

Reply:
Thanks, done!